pkg/registry/endpoint/strategy.go

```diff
@@ -46,10 +46,12 @@ func (endpointsStrategy) NamespaceScoped() bool {
 
 // PrepareForCreate clears fields that are not allowed to be set by end users on creation.
 func (endpointsStrategy) PrepareForCreate(ctx api.Context, obj runtime.Object) {
+	validation.ClearReadOnlyEndpointFieldsOnCreate(obj.(*api.Endpoints))
```

Comment:
We already have "ValidateEndpoints" and "ValidateEndpointsUpdate" functions - these should go there.